sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveContext.scala

```diff
@@ -272,6 +244,44 @@ class HiveContext(sc: SparkContext) extends SQLContext(sc) {
         Nil
     }
 
+  override protected[sql] def createSession(): SQLSession = {
+    new this.SQLSession()
+  }
+
+  protected[hive] class SQLSession extends super.SQLSession {
+    protected[sql] override lazy val conf: SQLConf = new SQLConf {
+      override def dialect: String = getConf(SQLConf.DIALECT, "hiveql")
+    }
+
+    protected[hive] lazy val hiveconf: HiveConf = {
```

Comment:
Any real use cases here? Maybe your authentication patch? In general, it's not recommended to use `HiveConf` as a mutable configuration collection.